meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc

```diff
@@ -16,7 +16,14 @@ do_mender_tar_src() {
     cd ${WORKDIR}
     # Zip up all plain files, plus the git directory, but not the rest, because
     # there are a lot of bitbake data directories.
-    tar czf ${TMPDIR}/mender-u-boot-src.tar.gz `find -maxdepth 1 -type f` git
+    if [ -n "${EXTERNALSRC}" ]; then
+        # tar u-boot sources, but transform ${S} as if it was git subdir in ${WORKDIR}
+        local tmp=${S}
+        tmp=${tmp#?}
+        tar --transform="s|${tmp}|git|" -czf ${TMPDIR}/mender-u-boot-src.tar.gz `find -maxdepth 1 -type f` ${S}
+    else
+        tar -czf ${TMPDIR}/mender-u-boot-src.tar.gz `find -maxdepth 1 -type f` git
+    fi
```

Comment:
Please use [the fix that was submitted to master](https://github.com/mendersoftware/meta-mender/commit/58b19eac9a48c44b875185573f4d8af86e23db01) so we don't end up with two different approaches. This should also be in a separate commit, not mixed with the machine type fixes.